packages/babel-helper-compilation-targets/src/index.js

```diff
@@ -170,30 +168,34 @@ type ParsedResult = {
 };
 
 export default function getTargets(
-  targets: Object = {},
+  inputTargets: InputTargets = {},
   options: Object = {},
 ): Targets {
   const targetOpts: Targets = {};
 
-  validateTargetNames(targets);
+  validateTargetNames(inputTargets);
 
   // `esmodules` as a target indicates the specific set of browsers supporting ES Modules.
   // These values OVERRIDE the `browsers` field.
-  if (targets.esmodules) {
+  if (inputTargets.esmodules) {
     const supportsESModules = browserModulesData["es6.module"];
-    targets.browsers = Object.keys(supportsESModules)
+    inputTargets.browsers = Object.keys(supportsESModules)
       .map(browser => `${browser} ${supportsESModules[browser]}`)
       .join(", ");
   }
 
   // Remove esmodules after being consumed to fix `hasTargets` below
-  delete targets.esmodules;
+  delete inputTargets.esmodules;
 
   // Parse browsers target via browserslist
-  const browsersquery = validateBrowsers(targets.browsers);
+  const browsersquery = validateBrowsers(inputTargets.browsers);
+  delete inputTargets.browsers;
+
+  // $FlowIgnore
+  let targets = (inputTargets: Targets);
```

Comment:
It was complaining a lot because `esmodules`/`browsers` are not part of `Targets`, which should contain the _resolved_ targets.